modules/apps/commerce/commerce-currency-service/src/main/java/com/liferay/commerce/currency/service/impl/CommerceCurrencyLocalServiceImpl.java

```diff
@@ -463,6 +441,22 @@ public void updateExchangeRates() throws PortalException {
 			ArrayUtil.toLongArray(commerceCurrencyFinder.getCompanyIds()));
 	}
 
+	@Activate
+	protected void activate(BundleContext bundleContext) {
```

Comment:
@JanisZhang Once you converted the `PortalInstanceLifecycleListenerImpl` to an OSGi component, you no longer need this service register and unregister logic. So, this class doesn't need the `activate` and `deactivate` method.